### ISSUE TITLE
Authoring content requires admin access

### ISSUE THREAD

**Issue by dmcassel** [marklogic-community/RunDMC#338]
I tried setting someone up with the dmc-user role, but he got errors. Set up the app so that having a sub-admin role is sufficient.

**Comment by mariannemyers** [comment]
How should the documents permissions be handled?  I submitted PR #362 that adds new roles - dmc-author and dmc-admin and uses these roles to control access to the publish/unpublish function.  Currently a dmc-author user will not see any documents that were previously inserted/updated by the admin since no document permissions exist (yet).

**Comment by mblakele** [comment]
https://github.com/marklogic/RunDMC/blob/docapp/src/model/data-access.xqy#L158 does a little prep work for this, commit 3db3698d6d1553f5477c53012a960f88d3e3ff41, by wrapping `xdmp:document-insert` calls in a library function for docs setup. However there are still a couple dozen calls elsewhere.

**Comment by mariannemyers** [comment]
Should I proceed further with adding document permissions, using commit 3db3698 as a guide?

**Comment by dmcassel** [comment]
@mariannemyers, sorry for the delayed response. Yes, please proceed. That commit looks like a good approach. It relies on xdmp:default-permissions(), which should work fine.

**Comment by mariannemyers** [comment]
Should only documents created/updated via the Admin UI be wrapped in a similar ml:document-insert?  My thought is to add a new module to src/admin/controller/modules/doucment-insert.xqy and edit appropriate calls to xdmp:document-insert under src/admin/controller to something similar to the ml:document-insert in commit 3db3698.  

I do not believe all xdmp:document-insert calls throughout the app should be changed, only the ones specifically in the src/admin/controller.

**Comment by dmcassel** [comment]
Yes, I agree with that. Please include the Media document insert.

On Fri, Jan 23, 2015 at 6:52 PM, Marianne Myers <notifications@github.com>
wrote:

> Should only documents created/updated via the Admin UI be wrapped in a
> similar ml:document-insert? My thought is to add a new module to
> src/admin/controller/modules/doucment-insert.xqy and edit appropriate calls
> to xdmp:document-insert under src/admin/controller to something similar to
> the ml:document-insert in commit 3db3698
> <https://github.com/marklogic/RunDMC/commit/3db3698d6d1553f5477c53012a960f88d3e3ff41>.
>
>
> I do not believe all xdmp:document-insert calls throughout the app should
> be changed, only the ones specifically in the src/admin/controller.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/marklogic/RunDMC/issues/338#issuecomment-71286852>.
>



-- 
http://blog.davidcassel.net
http://www.linkedin.com/in/davidcassel
